Automattic/wp-calypso
# Inline Help: Adding internal Calypso links to inline help search results.

## About this PR

This is an experimental feature that enables 'admin section' search results, that is, it surfaces sections of Calypso based on a search in the inline help field.

<img width="347" alt="Screen Shot 2019-04-02 at 5 03 45 pm" src="https://user-images.githubusercontent.com/6458278/55379594-4f304000-5569-11e9-91f5-0a1f21340539.png">

The _down-the-track-some-day-blue-sky_ idea is to create a stand alone filter library that might we employ in other features, for example, to filter the sidebar menu items:

<img width="459" alt="Screen Shot 2019-04-01 at 10 56 11 am" src="https://user-images.githubusercontent.com/6458278/55297148-cfbe4600-546e-11e9-888f-42f2587f9efb.png">

## Testing instructions

This feature is currently behind a development feature flag `help/admin-section-search`.

Open the inline help, making sure you have a site selected. For example, start here:

`http://calypso.localhost:3000/view/{yoursweettestsite}.wordpress.com`

Also search when no site is selected, e.g., `http://calypso.localhost:3000/me` 

You should be asked to select a site should you click on a section that requires one.

The list of admin sections is incomplete, but you can search for terms such as `'Add a domain'`, `'user'`, `'analytics'` and whichever other terms tickle your fancy, bearing in mind the [current definitions](https://github.com/Automattic/wp-calypso/pull/31903/files#diff-5cf1c6b9497fdc569ac7d61cb67e6958R18).

Clicking on a search results should also trigger a tracks event – `calypso_inlinehelp_admin_section_search` – with the following data:

```
{
    link: 'https://the-link-you-clicked-on.com',
    search_term: 'the search term',
}
```

## Notes

We might be able to extend and continue with #27173 (localizing the help text and links) after this PR. At the moment, help text and support doc links are in English.
